builtin/logical/pki/secret_certs.go

```diff
@@ -49,10 +51,36 @@ func (b *backend) secretCredsRevoke(ctx context.Context, req *logical.Request, _
 	defer b.revokeStorageLock.Unlock()
 
 	sc := b.makeStorageContext(ctx, req.Storage)
-	colonSerial := denormalizeSerial(serialInt.(string))
-	resp, err := revokeCert(sc, colonSerial, true)
-	if resp == nil && err == nil {
+
+	certEntry, err := fetchCertBySerial(sc, "certs/", serialInt.(string))
+	if err != nil {
+		switch err.(type) {
+		case errutil.UserError:
+			return logical.ErrorResponse(err.Error()), nil
```

Comment:
FWIW, I'm fairly sure we can't actually return a UserError here, so we could just return the error directly. If we're concerned, we could wrap the entire handler and do this one, which might be a best practice overall tbh, rather than continuing in this trend of doing it on a one-off basis.

Reply:
Yes you are correct, it can't be returned so I'll simplify this